ch_pipeline/analysis/flagging.py

```diff
@@ -1874,3 +1874,109 @@ def process(self, timestream):
         )
 
         return timestream
+
+
+class IterHPFMask(task.SingleTask):
+    """Generate a frequency mask by iteratively high-pass filtering the RMS of the
+    given data and masking the frequencies that exceed a noise threshold estimated
+    from adjacent frequency bins.
+
+    This task applies the algorithm implemented in `ch_util.rfi.iterative_hpf_masking`
+    to the RMS along a specified axis of the data provided. Flags are derived for every
+    element of the remaining axes, and combined at the end to generate a single frequency
+    mask.
+
+    Attributes
+    ----------
+    dataset: str
+        The dataset to use to derive the mask.
+    rms_axis: str
+        The axis to use to compute the RMS over.
+    niter: int
+        The maximum number of iterations.
+    delay_cut: float
+        The delay cut for the high-pass filter (in micro-seconds).
+    threshold: float
+        The threshold in units of noise standard deviation at which to flag
+        frequencies at every iteration.
+    min_frac_good: float
+        The minimum fraction of unflagged data at a given frequency required for it
+        not to be masked.
+    """
+
+    threshold = config.Property(proptype=float, default=5.0)
+    niter = config.Property(proptype=int, default=10)
+    delay_cut = config.Property(proptype=float, default=0.6)
+    rms_axis = config.Property(proptype=str, default="ra")
+    dataset = config.Property(proptype=str, default="map")
+    min_frac_good = config.Property(proptype=float, default=0.75)
+
+    def process(self, data):
+        """Generate a mask for the data.
+
+        Parameters
+        ----------
+        data: draco.containers.ContainerBase
+            Data to be used to generate the mask.
+        """
+
+        from mpi4py import MPI
+
+        if self.dataset not in data.datasets:
+            raise ValueError(f"Container {type(data)} has no '{self.dataset}' dataset.")
+        axes = list(data[self.dataset].attrs["axis"])
+        if self.rms_axis not in axes:
+            raise ValueError(
+                f"Dataset '{self.dataset}' has no '{self.rms_axis}' axis'."
+            )
+
+        # figure out what is the largest remaning axis
+        rem_axes = [a for a in axes if a not in ["freq", self.rms_axis]]
+        rem_ind = [axes.index(a) for a in rem_axes]
+        dist_axis = rem_axes[np.argmax([data[self.dataset].shape[i] for i in rem_ind])]
```

Comment:
Maybe more Pythonic, though maybe less clear? Not really required, I was just having fun.
```suggestion
        rem_axes = [(data[self.dataset].shape[ii], a) for ii, a in enumerate(axes) if a not in ["freq", self.rms_axis]]
        _, dist_axis = max(rem_axes)
```

Reply:
👍